app/src/main/java/com/android/shelfLife/model/camera/BarcodeScannerViewModel.kt

```diff
@@ -8,35 +8,63 @@ import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
 import androidx.core.content.ContextCompat
-import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.ViewModel
+import com.android.shelfLife.model.foodFacts.FoodFacts
+import com.android.shelfLife.model.foodFacts.FoodFactsRepository
+import com.android.shelfLife.model.foodFacts.SearchStatus
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+import kotlinx.coroutines.flow.StateFlow
 
 /**
  * ViewModel for managing the barcode scanner screen.
  *
  * @property application The application context
  */
-class BarcodeScannerViewModel(application: Application) : AndroidViewModel(application) {
+@HiltViewModel
+class BarcodeScannerViewModel
+@Inject
+constructor(
+    @ApplicationContext private val context: Context,
+    private val foodFactsRepository: FoodFactsRepository
+) : ViewModel() {
 
+  private val application = context as Application
   private val sharedPreferences =
-      getApplication<Application>()
-          .applicationContext
-          .getSharedPreferences("app_prefs", Context.MODE_PRIVATE)
+      application.getSharedPreferences("app_prefs", Context.MODE_PRIVATE)
 
   var permissionGranted by
       mutableStateOf(
-          ContextCompat.checkSelfPermission(
-              getApplication<Application>().applicationContext, Manifest.permission.CAMERA) ==
+          ContextCompat.checkSelfPermission(context, Manifest.permission.CAMERA) ==
               PackageManager.PERMISSION_GRANTED)
     private set
 
   private var permissionRequested by
       mutableStateOf(sharedPreferences.getBoolean("permissionRequested", false))
 
+  val searchStatus: StateFlow<SearchStatus> = foodFactsRepository.searchStatus
+
+  val foodFactsSuggestions: StateFlow<List<FoodFacts>> = foodFactsRepository.foodFactsSuggestions
+
+  init {
+    if (!permissionRequested) {
+      checkCameraPermission()
+    }
+  }
+
+  fun searchByBarcode(barcode: Long) {
+    foodFactsRepository.searchByBarcode(barcode)
+  }
+
+  fun resetSearchStatus() {
```

Comment:
you add this functions that allows to reset the search status. This function is used to connect the front end and back end